browser/src/map/Map.js

```diff
@@ -424,6 +453,37 @@ L.Map = L.Evented.extend({
 	},
 
 	initializeModificationIndicator: function() {
+		this._docStatusUI = document.getElementById('documentstatus');
+		if (this._docStatusUI !== null && this._docStatusUI !== undefined) {
```

Comment:
would be good to extract that widget into separate file (not Map.js)

it's inserted to Statusbar as `customitem` we can create "widget factory method" like other widgets
also there is mechanism to update the widget there

Reply:
for update: please see example of `onPageChange(e)` in StatusBar.js
for creation of html for a widget: `this._generateHtmlItem('statepagenumber'),`
which produces widget of type `htmlcontent` and it's content is generated in https://github.com/CollaboraOnline/online/blob/master/browser/src/control/jsdialog/Widget.HTMLContent.ts#L121

so would be good to follow the same approach as other widgets and encapsulate a bit

Reply:
:) This is funny. My first implementation was exactly as you are suggesting. However, I didn't like that updateWidget recreates the whole content of the widget every time there is a change in its content. But if this is the approach that you prefer then I'm OK with it. Just a question: should I also use updateWidget every time the "Last saved" information changes or I can save the element in a data member and just change the text (as it is now) ? Of course, I will use updateWidget for all the other cases (like switching from "Saving..." to "Document Saved").

Reply:
> would be good to extract that widget into separate file (not Map.js)
> 
> it's inserted to Statusbar as `customitem` we can create "widget factory method" like other widgets also there is mechanism to update the widget there

A completely new file ?

Reply:
> > would be good to extract that widget into separate file (not Map.js)
> A completely new file ?
no, it's fine just to move to the other files

Reply:
> :) This is funny. My first implementation was exactly as you are suggesting. However, I didn't like that updateWidget recreates the whole content of the widget every time there is a change in its content. But if this is the approach that you prefer then I'm OK with it. Just a question: should I also use updateWidget every time the "Last saved" information changes or I can save the element in a data member and just change the text (as it is now) ? Of course, I will use updateWidget for all the other cases (like switching from "Saving..." to "Document Saved").

I also don't like that, but for now it will be the best, as later we can rework all of them at once.
The reason is: we have rules of how our "JSDialog" framework works, any hacks to optimize something may lead to worse state.
We have "actions" which can do just a text modification, but let's not overcomplicate now. We need to rework all of the HTMLContent widgets anyway (as it is just an adapter for legacy widgets to JSDialogs)